### PR TITLE
Remove models from gridt_server/resources files

### PR DESCRIPTION
**Pull request by mehfluffy**
Partially solves Issue #93.

Fixed files in [gridt_server/resources](https://github.com/GridtNetwork/gridt-server/tree/GridtLibIncorporation/web/gridt_server/resources) by removing their dependency on `gridt_server.models`.

In helpers.py, deleted functions that depended on `gridt_server.models` but were never used anywhere else. 

In user.py, also deleted other unnecessary imports namely `jwt` and `util.email_templates` (lines 15-19). After using `gridt.controllers` instead (in the class `ResetPasswordResource`), all the above imports were no longer needed and were deleted.

### CODE REVIEW

**Comment by Drvanon** [comment]
You have seemed to have removed the right stuff. Now please also remove the stuff from the tests.

File: web/gridt_server/tests/unit/test_schemas.py [hunk 1]
```diff
@@ -1,152 +0,0 @@
...
-    RequestEmailChangeSchema,
-    ChangeEmailSchema,
-)
-
-from freezegun import freeze_time
-from gridt_server.models.user import User
-
-
-class SchemasTest(BaseTest):
-    def test_movement_schema_long(self):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/unit/test_schemas.py:16]: Most tests here do not use models.

File: web/gridt_server/tests/unit/test_schemas.py [hunk 2]
```diff
@@ -1,152 +0,0 @@
...
-                error.exception.messages,
-                {
-                    "short_description": ["Length must be between 10 and 100."],
-                    "name": ["Length must be between 4 and 50."],
-                },
-            )
-
-    def test_request_email_change_schema_invalid(self):
-        with self.app_context():
-            user = self.create_user()
```
Comment (PJvanVliet) [inline web/gridt_server/tests/unit/test_schemas.py:77]: This test does use models.

File: web/gridt_server/tests/unit/test_schemas.py [hunk 3]
```diff
@@ -1,152 +0,0 @@
...
-
-            schema = ChangeEmailSchema()
-            with self.assertRaises(ValidationError) as error:
-                schema.load(bad_request)
-
-                self.assertEqual(
-                    error.exception.messages, {"token": ["Invalid token."]}
-                )
-
-    def test_change_email_schema_correct(self):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/unit/test_schemas.py:142]: This test also uses models

File: web/gridt_server/tests/unit/test_schemas.py [hunk 4]
```diff
@@ -1,152 +0,0 @@
...
-                "new_email": "proper@email.com",
-            }
-            schema = RequestEmailChangeSchema()
-            schema.context = {"user": user}
-
-            # Make sure no error is thrown with this info
-            res = schema.load(proper_request)
-            self.assertEqual(res, proper_request)
-
-    def test_change_email_schema_token_expired(self):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/unit/test_schemas.py:109]: This test also uses models.

File: web/gridt_server/tests/unit/test_schemas.py [hunk 5]
```diff
@@ -1,152 +0,0 @@
...
-
-                self.assertEqual(
-                    error.exception.messages,
-                    {
-                        "password": ["Failed to identify user with given password."],
-                        "new_email": ["Not a valid email address."],
-                    },
-                )
-
-    def test_request_email_change_schema_correct(self):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/unit/test_schemas.py:95]: So does this one.

File: web/gridt_server/tests/integration/resources/test_register.py [hunk 6]
```diff
@@ -1,38 +0,0 @@
-import json
-from gridt_server.tests.base_test import BaseTest
-from gridt_server.models.user import User
-
-
-class RegistrationResourceTest(BaseTest):
-    def test_registration(self):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/integration/resources/test_register.py:7]: This test does not use models.

File: web/gridt_server/tests/base_test.py [hunk 7]
```diff
@@ -1,172 +0,0 @@
...
-
-
-class LoggedTestCase(TestCase):
-    __metaclass__ = LogThisTestCase
-    logger = logging.getLogger()
-    # Uncomment below to enable logging while testing
-    # logger.setLevel(logging.DEBUG)
-
-
-class BaseTest(LoggedTestCase):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/base_test.py:62]: You'll want to keep this. Without it, we cannot run tests. There are a few functions that are not able to remain due to their reliance on models.

File: web/gridt_server/tests/base_test.py [hunk 8]
```diff
@@ -1,172 +0,0 @@
...
-        self.app_context = app.app_context
-
-        self.users = []
-        self.movements = []
-
-        engine = create_engine(app.config["SQLALCHEMY_DATABASE_URI"])
-        Session.configure(bind=engine)
-        Base.metadata.create_all(engine)
-
-    def tearDown(self):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/base_test.py:82]: All functions from here down rely on the database / models. They can no longer be used.

File: web/gridt_server/resources/user.py [hunk 9]
```diff
@@ -96,15 +90,7 @@ class ResetPasswordResource(Resource):
...
-        token_decoded = jwt.decode(data["token"], secret_key, algorithms=["HS256"])
-        user_id = token_decoded["user_id"]
-        password = data["password"]
-
-        user = User.find_by_id(user_id)
-        user.hash_and_store_password(password)
-
-        send_password_change_notification(user.email)
+        secret_key = current_app.config["SECRET_KEY"]
+        reset_password(data["token"], data["password"], secret_key)
```
Comment (PJvanVliet) [inline web/gridt_server/resources/user.py:95]: This indeed seems the correct way to call the gridtlib.

**Review by PJvanVliet — CHANGES_REQUESTED** [review]
On the whole, I think a little bit too much was removed. Before merging, you should probably restore the tests that do not explicitly use models.

File: web/gridt_server/tests/unit/test_schemas.py (hunk 2 above)
Comment (PJvanVliet) [reply web/gridt_server/tests/unit/test_schemas.py:77]: Forgive my sloppy review. Here, I mean that removing this is fine. In SchemasTest, you should probably restore the tests that do not rely on models.

**Comment by mehfluffy** [comment]
Indeed I have removed way too many things, I have now reset my head to 6c5a1fe and branched from there to work on removing models from the tests and rewriting them for Issue #97.

**Comment by mehfluffy** [comment]
Since I didn't do much for Issue #97 yet, I will keep them somewhere and now focus on restoring the files and deleting the tests that @PJvanVliet mentioned on branch 93.

File: web/gridt_server/tests/base_test.py (hunk 7 above)
Comment (PJvanVliet) [reply web/gridt_server/tests/base_test.py:62]: Actually, all functions here rely on models.

**Comment by mehfluffy** [comment]
With ada196e I also updated the big comment on Issue #93 with info on the tests/integration/models and tests/unit/models tests

File: web/gridt_server/tests/unit/test_schemas.py [hunk 10]
```diff
@@ -9,7 +9,6 @@
 )
 
 from freezegun import freeze_time
-from gridt_server.models.user import User
 
 
 class SchemasTest(BaseTest):
```
Comment (PJvanVliet) [inline web/gridt_server/tests/unit/test_schemas.py:14]: We should fix this before moving on. We should keep tests that do not use models!

**Review by PJvanVliet — CHANGES_REQUESTED** [review]

**Review by PJvanVliet — APPROVED** [review]

**Pull request closed without merging**